### ISSUE TITLE
Wrong parsed date with time at 24:00:00

### ISSUE THREAD

**Issue by jRichardeau** [date-fns/date-fns#1228]
Hey,

Using date-fns-2.0.0-beta.1, it seems to not parse date correctly if there is a time like 24:00:00.

`new Date("2000-01-01T24:00:00.0")` => Jan 02 2000 00:00:00
`parse("2000-01-01T24:00:00.0")` => Jan 01 2000 00:00:00

Thank you

**Comment by kossnocorp** [comment]
Please provide the full example of how you use it, what's the format string, what's the output in local time and what is the backup date (the third argument).

**Comment by jRichardeau** [comment]
Sorry it's `parseISO` not `parse`, so no more argument.

The output in local time is : 

`new Date("2000-01-01T24:00:00.0")` => Sun Jan 02 2000 00:00:00 GMT-0500
`parseISO("2000-01-01T24:00:00.0")` => Sat Jan 01 2000 00:00:00 GMT-0500

**Comment by kossnocorp** [comment]
I see, then it's actually the correct result while the value produced by `new Date` is incorrect. [According to the ISO 8601](https://en.wikipedia.org/wiki/ISO_8601#Times):

> [hh] refers to a zero-padded hour between 00 and 24 (where 24 is only used to denote midnight at the end of a calendar day).
> ...
> Midnight is a special case and may be referred to as either "00:00" or "24:00". The notation "00:00" is used at the beginning of a calendar day and is the more frequently used. At the end of a day use "24:00". "2007-04-05T24:00" is the same instant as "2007-04-06T00:00" (see Combined date and time representations below).

So that is essentially represents the 0 of the next day. For example, if you use `Date.parse` you get the same result:

```js
new Date(Date.parse("2000-01-01T24:00:00.0"))
//=> Sun Jan 02 2000 00:00:00 GMT+0400 (Samara Standard Time)
```

Exactly because of such inconsistencies in `new Date()` behavior, we dropped strings support in favor of implicit parsing.

May I ask where from such dates (24:00) are coming from? What language/environment generate those?

If you don't control it and can't change the output, I would recommend using `new Date()` instead of `parseISO`.

**Comment by kossnocorp** [comment]
Yes, you're right it's incorrect behavior. We'll fix that. Thank you for reporting!

**Comment by jRichardeau** [comment]
Thank you !

**Comment by leshakoss** [comment]
Was fixed in `v2.0.0-beta.2`.

**Comment by kossnocorp** [comment]
It was fixed in `v2.0.0-beta.2`. Check it out!
